cmd/opampsupervisor/specification/README.md

```diff
@@ -220,19 +225,60 @@ telemetry:
 
 ```
 
-**Note:**
+#### Notes on `agent::config_files`, `agent::args`, and `agent::env`
+
+Please be aware that when using the `agent::config_files` parameter,
+the configuration files specified are applied in the order they are specified.
+In other words, configuration files are merged from the top of the list to the bottom.
+Configuration added by files at the top of the list may be overwritten by the later ones.
+
+The indicated configuration files are merged in memory and the resulting configuration
+is written to `<storage::directory>/effective.yaml`.
 
-Please be aware that when using the `.agent.config_files` parameter,
-the configuration files specified are applied after the configuration from the OpAMP server.
-After the configuration files, arguments present in `.agent.args` are passed to the executable binary.
-The environmanet variables specified in `.agent.env` are set in the collector process environment.
+There are a few "special" configuration files that can be used to completely
+customize final configuration given to the Collector. Below are the available
+values and what they represent:
 
-The following configuration:
+- `$OPAMP_EXTENSION_CONFIG`: configuration for the OpAMP extension to connect to the Supervisor.
+- `$OWN_TELEMETRY_CONFIG`: configuration for the Collector to report its own metrics.
+- `$BUILTIN_CONFIG`: built-in base configuration.
+- `$REMOTE_CONFIG`: remote configuration received by the Supervisor.
+
+**NOTE**: These configuration snippets, particularly `$OPAMP_EXTENSION_CONFIG`, are essential for the Supervisor and Collector to work together. Overriding values in these may result in the Supervisor failing to properly start the Collector and should be done with caution.
+
+These special files can be mixed with user-provided configuration files to create complex
+configuration merge orders, for instance, creating base-layer configuration at the
+lowest priority while keeping compliance configuration at the highest priority:
+
+```yaml
+agent:
+  config_files:
+    - base_config.yaml
+    - $OWN_TELEMETRY_CONFIG
+    - $BUILTIN_CONFIG
```

Comment:
Sorry to add this so late, but I think we should merge these files now that we're exposing them to users: looking at them, they both configure the Collector's telemetry, and I think we should just put them together if users need to understand what each is.

We could do this in a pre-req PR or in this PR, I'm fine either way.

Reply:
The urgency for doing this as a prerequisite is that it will be a breaking change if we expose these options then change them, so we should just get it in now.

Reply:
That's a good idea. Let me see how big would the change be. If it's small enough I will include it here.

Reply:
Whoops, sorry, I probably should have specified: could you call the resulting file `OWN_TELEMETRY_CONFIG`? I think that's the most accurate description of it right now.